### PR TITLE
feat: allow use of OpenSCAP result files in task xccdf_result_to_oscal_ar

### LINKED ISSUE #1410: Allow the use of OpenSCAP files in task xccdf-result-to-oscal-ar
## Issue description / feature objectives
Currently if you pass an OpenSCAP result file to the xccdf-result-to-oscal-ar task the resulting OSCAL file is blank.
## Caveats / Assumptions
It may may be wise to try this with other scanners like NIWC's SCC. 
## Completion Criteria
Using OpenSCAP result files will result in populated OSCAL AR files.

### PR DESCRIPTION
**Pull request by Mab879**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [ ] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [ ] My PR meets testing requirements.
- [ ] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Before this commit if you wanted to use result files from OpenSCAP in the task xccdf_result_to_oscal_ar you had to extract the `TestResult` element and place it as the root of the XML document, otherwise the resulting OSCAL document would be blank. Thus making it impossible to directly use output from OpenSCAP with the task.

With this commit the task will detect that the root element is not `TestResult` and then it will find the `TestResult` element in the XML document. This allows the use of files created by OpenSCAP using the `--results` and `--results-arf` switches.

Fixes #1410

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

**Review by degenaro — COMMENTED** [review]
Thanks for this submission!

1. Are other namespaces possible? If so should there be a list used until the right one is found?
2. Alternatively, should the `trestle task xccdf-result-to-oscal-ar` be enhanced to accept an optional config flag, perhaps `namespace = http://checklists.nist.gov/xccdf/1.2`?
3. Can you add a test case to `tests/trestle/tasks/xccdf_result_to_oscal_ar_test.py`?

**Comment by Mab879** [comment]
>     1. Are other namespaces possible? If so should there be a list used until the right one is found?

There might be one more, I can add that to the task logic.

>     2. Alternatively, should the `trestle task xccdf-result-to-oscal-ar` be enhanced to accept an optional config flag, perhaps `namespace = http://checklists.nist.gov/xccdf/1.2`?

I think searching for the correct namespace might be best option.

>     3. Can you add a test case to `tests/trestle/tasks/xccdf_result_to_oscal_ar_test.py`?

Done. The test files are rather large, I welcome suggestions on this.

**Comment by degenaro** [comment]
With respect to the code: is `http://checklists.nist.gov/xccdf/1.2` the only possibility or are there other common versions that should be checked if 1.2 is not the correct one (e.g. 1.1, 1.3, 2.0...)?

With respect to the test cases, please consider: it would seem that only 1 new test case and 1 corresponding data file are needed to cover the new code, and the test data file should be reduced (hacked, if you will) to the bare minimum amount needed to perform the new ns test.

Thx!

**Comment by degenaro** [comment]
Do you wish to continue with this PR? Another release of trestle is imminent, then the direction will be toward upgrading to the current version of OSCAL.

Also, the value for `ns` should not be hardcoded? Nor should the search term `.//checklist12:TestResult` ?

**Comment by Mab879** [comment]
This is something that I'm still interested in working on.

The `ns` value is the XML namespace mapping and `checklist12:TestResult` is just using those XML namespaces.

**Comment by degenaro** [comment]
> This is something that I'm still interested in working on.
> 
> The `ns` value is the XML namespace mapping and `checklist12:TestResult` is just using those XML namespaces.

But it is bad practice to hardocde values in code that are not typically universal, no?

**Comment by Mab879** [comment]
> > This is something that I'm still interested in working on.
> > The `ns` value is the XML namespace mapping and `checklist12:TestResult` is just using those XML namespaces.
> 
> But it is bad practice to hardocde values in code that are not typically universal, no?

So the `ns` values are from the standards so the should be universal.

**Comment by degenaro** [comment]
Please try running:
```
make code-format
make code-lint
```

**Comment by degenaro** [comment]
We are close to making a release in the next few days. If you can run the make commands above and deliver the revised formatted code, I can approve, then you can merge. Otherwise there will be a freeze on new features until after upgrading to new version of OSCAL 1.1.2. Thx!

**Comment by degenaro** [comment]
tests/trestle/tasks/xccdf_result_to_oscal_ar_test.py:449:1: D103 Missing docstring in public function
tests/trestle/tasks/xccdf_result_to_oscal_ar_test.py:470:1: D103 Missing docstring in public function

**Comment by Mab879** [comment]
> tests/trestle/tasks/xccdf_result_to_oscal_ar_test.py:449:1: D103 Missing docstring in public function tests/trestle/tasks/xccdf_result_to_oscal_ar_test.py:470:1: D103 Missing docstring in public function

Thanks for the output. I have struggling to get `make code-lint` to run locally.

**Comment by degenaro** [comment]
Here is the trick: 
1. python -m venv venv.trestle
2. source venv.trestle/bin/activate
3. cd compliance-trestle
4. make install
5. make develop
6. make code-format
7. make code-lint

**Review by degenaro — APPROVED** [review]
LGTM.

Please:
1. Update Branch (out of sync)
2. Squash and Merge

**Comment by Mab879** [comment]
> LGTM.
> 
> Please:
> 
>     1. Update Branch (out of sync)
> 
>     2. Squash and Merge

Done. Looks like I will need approval again for the CI to run.

**Comment by degenaro** [comment]
I think you should be able to squash and merge now...?

**Comment by Mab879** [comment]
> I think you should be able to squash and merge now...?

I'm still seeing "Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests."

**Comment by degenaro** [comment]
Ugh. I did the squash and merge, no worries. Thanks for this contribution!

**Pull request merged**
